doc/classes/NavigationAgent.xml

```diff
@@ -0,0 +1,163 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="NavigationAgent" inherits="Node" category="Core" version="3.2">
+	<brief_description>
+		3D Agent used in navigation for collision avoidance.
+	</brief_description>
+	<description>
+		3D Agent that is used in navigation to reach a location while avoiding static and dynamic obstacles. The dynamic obstacles are avoided using RVO collision avoidance. The agent needs navigation data to work correctly. This can be done by having the agent as a child of a [Navigation] node, or using [method set_navigation].
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="distance_to_target" qualifiers="const">
+			<return type="float">
+            </return>
+			<description>
+				Returns the distance to the target location, using the agent's global position. The user must set the target location with [method set_target_location] in order for this to be accurate.
+			</description>
+		</method>
+		<method name="get_final_location">
+			<return type="Vector3">
+			</return>
+			<description>
+				Returns the reachable final location in global coordinates. This can change if the navigation path is altered in any way.
```

Comment:
This can change by time to time if the navigation map change, so it is more appropriate if you check this each frame.